### PR TITLE
Cancel archive New endpoint

### PR DESCRIPTION
**Pull request by lrivera-oicr**

### CODE REVIEW

File: src/main/java/org/cancogenvirusseq/singularity/api/ApiDefinition.java [hunk 1]
```diff
@@ -155,4 +154,20 @@ public interface ApiDefinition {
...
   Mono<Archive> buildArchiveWithSetId(@RequestBody SetIdBuildRequest setIdBuildRequest);
+
+  @ApiOperation(
+    value = "Cancel Build process of given hashes",
+    nickname = "Cancel Build",
+    tags = "Singularity API")
+  @ApiResponses(
+    value = {
+      @ApiResponse(code = 200, message = "", response = CancelListResponse.class),
+      @ApiResponse(code = 500, message = UNKNOWN_MSG, response = ErrorResponse.class)
```
Comment (caravinci) [inline src/main/java/org/cancogenvirusseq/singularity/api/ApiDefinition.java:165]: out of curiosity, where is this UNKNOWN_MSG defined, and what does it contain?
is it spring boot or flux magic?

File: src/main/java/org/cancogenvirusseq/singularity/api/model/HashResult.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class HashResult {
+  private String hash;
+  private String oldStatus;
+  private String newStatus;
+  private String createdAt;
```
Comment (caravinci) [inline src/main/java/org/cancogenvirusseq/singularity/api/model/HashResult.java:16]: We could use also having the `type` and `numberOfSamples` here

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -56,3 +56,6 @@ intervalEventEmitter:
 
 utils:
   tsvListSeparator: ";"
+
+archive:
+  cancelPeriodSeconds: 14400
```
Comment (caravinci) [inline src/main/resources/application.yaml:61]: 🙌

File: src/main/java/org/cancogenvirusseq/singularity/repository/ArchivesRepo.java [hunk 4]
```diff
@@ -27,6 +28,10 @@ Mono<Archive> findTopByTypeAndStatusOrderByCreatedAtDesc(
 
   Mono<Archive> findArchiveByHashInfoEquals(String hashInfo);
 
+  Flux<Archive> findByHashInAndStatus(List<String> hash, ArchiveStatus status);
```
Comment (caravinci) [inline src/main/java/org/cancogenvirusseq/singularity/repository/ArchivesRepo.java:31]: `Hash In and Status`? what's `In` in this context?

**Review by caravinci — CHANGES_REQUESTED** [review]
Just a couple comments, and the additional object details in the response

File: src/main/java/org/cancogenvirusseq/singularity/api/ApiDefinition.java (hunk 1 above)
Comment (lrivera-oicr) [reply src/main/java/org/cancogenvirusseq/singularity/api/ApiDefinition.java:165]: it's an instance variable in ApiDefinition.java
String UNKNOWN_MSG = "An unexpected error occurred.";
it's used by @ApiResponse annotation (a swagger annotation) to describe the API response, it doesn't impact API functionality

Comment (caravinci) [reply src/main/java/org/cancogenvirusseq/singularity/api/ApiDefinition.java:165]: thanks!

File: src/main/java/org/cancogenvirusseq/singularity/repository/ArchivesRepo.java (hunk 4 above)
Comment (lrivera-oicr) [reply src/main/java/org/cancogenvirusseq/singularity/repository/ArchivesRepo.java:31]: "Hash In" is to specify multiple values in a WHERE clause.  
similar to say `"WHERE hash IN (...) AND status=..."`

Comment (caravinci) [reply src/main/java/org/cancogenvirusseq/singularity/repository/ArchivesRepo.java:31]: That makes sense! in that case, I think it'd be sufficiently clear as `findByHashAndStatus`.
Do you want to change it that way, please?

Comment (lrivera-oicr) [reply src/main/java/org/cancogenvirusseq/singularity/repository/ArchivesRepo.java:31]: as query is created based on the method's name, the `In` keyword is useful to query DB passing a list of hashes.
The suggested method name  `findByHashAndStatus` will create a query to look for a single hash.

File: src/main/java/org/cancogenvirusseq/singularity/api/model/HashResult.java (hunk 2 above)
Comment (lrivera-oicr) [reply src/main/java/org/cancogenvirusseq/singularity/api/model/HashResult.java:16]: `type` and `numberOfSamples` added

File: src/main/java/org/cancogenvirusseq/singularity/repository/ArchivesRepo.java (hunk 4 above)
Comment (caravinci) [reply src/main/java/org/cancogenvirusseq/singularity/repository/ArchivesRepo.java:31]: ah, lol I misread this code as a method's definition... not as an instantiation of a library's method.
My bad! please disregard 💯

**Review by caravinci — APPROVED** [review]
LGTM! Ready to "squash and merge"!

**Pull request merged**